frontend/alert/src/components/ManageAlertLogic.js

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from "react";
+import { ManageAlert } from "./managealert/ManageAlertUI";
+import { AlertStatus, AlertRepeat, AlertAction } from "./managealert/AlertItemEnums";
+
+const fetchAlerts = () => {
+    return fetch("/api/alert/api/registrationhistory")
+        .then(res => res.json());
+};
+
+const getStatusPromise = (sectionId) => {
+    return fetch(`/api/courses/current/sections/${sectionId}`);
+};
+
+const processAlerts = (setAlerts) => {
+    fetchAlerts()
+        .then((res) => {
+            // res is an array of JSON objects, each with information about
+            // an alert
+
+            // Initialize a Map mapping each section ID to a
+            // boolean, which represents whether a section is open
+            const sectionStatus = new Map();
+
+            // We only want to query each section's status once.
+            let sections = new Set();
+            res.forEach((section) => {
+                sections.add(section.section);
+            });
+            sections = Array.from(sections);
+
+            const promises = [];
+            sections.forEach((section) => {
+                promises.push(getStatusPromise(section));
+            });
+
+            Promise.all(promises)
+                .then(vals => Promise.all(vals.map(val => val.json())))
```

Comment:
Could this request happen inside of the `AlertItem` component? Since it's tied directly to that component. Might make this code cleaner to read / shorter.